### PR TITLE
Feature/us8234 group search in finder

### PR DESCRIPTION
**Pull request by vchtelmakh**

### CODE REVIEW

File: CI/SQL/20170518_131422_US8234-small-groups-in-finder.sql [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+		EXEC(@DynamicQuery)
+
+	END
+	END
+
+END
+
+
+GO
+
```
Comment (JoeKer1) [inline CI/SQL/20170518_131422_US8234-small-groups-in-finder.sql]: Can you show me what changed from what was out there? It's hard to tell. Also make sure old group tool still works.

File: Gateway/crds-angular/Services/FinderService.cs [hunk 2]
```diff
@@ -231,25 +238,25 @@ private void GatheringValidityCheck(int contactId, AddressDTO address)
```
Comment (JoeKer1) [inline Gateway/crds-angular/Services/FinderService.cs]: Right now this method goes and gets all the groups and that seems wasteful. Can we create a new method that gets a list of groups filtered by this data already?

File: Gateway/crds-angular/Services/FinderService.cs [hunk 3]
```diff
@@ -348,42 +355,61 @@ public int GetLeaderParticipantIdFromGroup(int groupId)
...
+            int id = _anywhereGroupType; 
+
+            if (finderType.Equals(_finderConnect))
             {
-                pin.Title = GetPinTitle(pin);
-                pin.IconUrl = GetPinUrl(pin.PinType);
-                //calculate proximity for all pins to origin
-                if (pin.Address.Latitude == null) continue;
-                if (pin.Address.Longitude != null) pin.Proximity = GetProximity(originCoords, new GeoCoordinate(pin.Address.Latitude.Value, pin.Address.Longitude.Value));
+                var cloudReturn = _awsCloudsearchService.SearchConnectAwsCloudsearch("matchall",
```
Comment (JoeKer1) [inline Gateway/crds-angular/Services/FinderService.cs]: nitpicky but we could encapsulate these in config

File: Gateway/crds-angular/Services/FinderService.cs [hunk 4]
```diff
@@ -348,42 +355,61 @@ public int GetLeaderParticipantIdFromGroup(int groupId)
...
+                                                                                     "_all_fields",
+                                                                                     _configurationWrapper.GetConfigIntValue("ConnectDefaultNumberOfPins"),
```
Comment (JoeKer1) [inline Gateway/crds-angular/Services/FinderService.cs]: Not sure if "ConnectDefaultNumberOfPins is used anywhere else. Probably should just set a constant in the constructor so we can use it where ever.

File: Gateway/crds-angular/Services/FinderService.cs [hunk 5]
```diff
@@ -348,42 +355,61 @@ public int GetLeaderParticipantIdFromGroup(int groupId)
...
             }
 
+            this.AddPinMetaData(pins, originCoords);
             return pins;
         }
 
         private string GetPinTitle(PinDto pin)
         {
-            string jsonData="";
+            string jsonData = "";
```
Comment (JoeKer1) [inline Gateway/crds-angular/Services/FinderService.cs]: Why are we doing this in json? Should this be in a class and then we let the serializer do ti?

File: Gateway/crds-angular/Services/FinderService.cs [hunk 6]
```diff
@@ -436,27 +464,27 @@ private List<PinDto> ConvertFromAwsSearchResponse(SearchResponse response)
...
-                    
+
                     pin.Gathering = new GroupDTO
                     {
                         GroupId = hit.Fields.ContainsKey("groupid") ? Convert.ToInt32(hit.Fields["groupid"].FirstOrDefault()) : 0,
                         GroupName = hit.Fields.ContainsKey("groupname") ? hit.Fields["groupname"].FirstOrDefault() : null,
                         GroupDescription = hit.Fields.ContainsKey("groupdescription") ? hit.Fields["groupdescription"].FirstOrDefault() : null,
                         PrimaryContactEmail = hit.Fields.ContainsKey("primarycontactemail") ? hit.Fields["primarycontactemail"].FirstOrDefault() : null,
-                        Address = pin.Address, 
+                        Address = pin.Address,
```
Comment (JoeKer1) [inline Gateway/crds-angular/Services/FinderService.cs]: Should this be pin.Gathering.Address?

File: Gateway/crds-angular/Services/FinderService.cs [hunk 7]
```diff
@@ -466,43 +494,43 @@ private List<PinDto> ConvertFromAwsSearchResponse(SearchResponse response)
...
 
         private static double Proximity(double lat1, double lon1, double lat2, double lon2)
         {
             var theta = lon1 - lon2;
-            var dist = Math.Sin(Deg2Rad(lat1)) * Math.Sin(Deg2Rad(lat2)) + Math.Cos(Deg2Rad(lat1)) * Math.Cos(Deg2Rad(lat2)) * Math.Cos(Deg2Rad(theta));
+            var dist = Math.Sin(Deg2Rad(lat1))*Math.Sin(Deg2Rad(lat2)) + Math.Cos(Deg2Rad(lat1))*Math.Cos(Deg2Rad(lat2))*Math.Cos(Deg2Rad(theta));
             dist = Math.Acos(dist);
             dist = Rad2Deg(dist);
-            dist = dist * 60 * 1.1515;
+            dist = dist*60*1.1515;
```
Comment (JoeKer1) [inline Gateway/crds-angular/Services/FinderService.cs]: Magic, move these to configs or const?

File: Gateway/crds-angular/Services/FinderService.cs [hunk 8]
```diff
@@ -763,5 +778,73 @@ private void SendGatheringInviteResponseEmail(bool inviteAccepted, PinDto host,
...
+
+        private List<PinDto> TransformGroupDtoToPinDto(List<GroupDTO> groupDTOs, string finderType)
+        {
+            var pins = new List<PinDto>();
+
+            if (finderType.Equals(_finderConnect))
+            {
+                foreach (var group in groupDTOs)
+                {
+                    var pin = Mapper.Map<PinDto>(group);
```
Comment (JoeKer1) [inline Gateway/crds-angular/Services/FinderService.cs]: If we take care of the TODO we might be able to setup automapper to do this for us. Probably still the same just less code.  Maybe, the pin type might not let us do that.

File: Gateway/crds-angular/Services/FinderService.cs (hunk 2 above)
Comment (JoeKer1) [inline Gateway/crds-angular/Services/FinderService.cs]: Could even just return the count directly from the rest api.

File: CI/SQL/20170518_131422_US8234-small-groups-in-finder.sql (hunk 1 above)
Comment (vchtelmakh) [inline CI/SQL/20170518_131422_US8234-small-groups-in-finder.sql]: I just tested search on int.crossroads.net/groups - it works.

File: Gateway/crds-angular/Services/FinderService.cs (hunk 7 above)
Comment (vchtelmakh) [inline Gateway/crds-angular/Services/FinderService.cs]: I changed them to constants, but they are pretty magical anyway. I have to assume someone just copied formulas from http://stackoverflow.com/questions/389211/geospatial-coordinates-and-distance-in-kilometers

File: Gateway/crds-angular/Services/FinderService.cs (hunk 6 above)
Comment (vchtelmakh) [inline Gateway/crds-angular/Services/FinderService.cs]: Not as far as I can tell - data comes back flat from AWS, we want the address on the top level.

File: Gateway/crds-angular/Services/FinderService.cs (hunk 5 above)
Comment (vchtelmakh) [inline Gateway/crds-angular/Services/FinderService.cs]: We need this as JSON on the front end to draw the labels. Its convoluted and dirty, but in short, we have a title attribute generated by NG2 Goog maps that we need to stuff label data into. We can't pass an object to the title, so we pass a JSON string. This causes parsing errors if group names have apostrophes, etc. We have a story to handle this.

File: CI/SQL/20170518_131422_US8234-small-groups-in-finder.sql (hunk 1 above)
Comment (JoeKer1) [inline CI/SQL/20170518_131422_US8234-small-groups-in-finder.sql]: 👍

**Pull request merged**